contracts/token/ERC721/ERC721BasicToken.sol

```diff
@@ -0,0 +1,279 @@
+  }
+
+  /**
+  * @dev Internal function to burn a specific token
+  * @dev Reverts if the token does not exist
+  * @param _tokenId uint256 ID of the token being burned by the msg.sender
+  */
+  function doBurn(address _owner, uint256 _tokenId) internal {
+    clearApproval(_owner, _tokenId);
+    removeToken(_owner, _tokenId);
+    Transfer(_owner, 0x0, _tokenId);
+  }
+
+  /**
+  * @dev Internal function to clear current approval and transfer the ownership of a given token ID
+  * @param _from address which you want to send tokens from
+  * @param _to address which you want to transfer the token to
+  * @param _tokenId uint256 ID of the token to be transferred
+  * @param _data bytes data to send along with a safe transfer check
+  * @param _safe bool whether to perform a safe transfer
+  */
+  function clearApprovalAndTransfer(address _from, address _to, uint256 _tokenId, bytes _data, bool _safe) internal {
+    require(_from != address(0));
+    require(_to != address(0));
+    require(_to != ownerOf(_tokenId));
+
+    clearApproval(_from, _tokenId);
+    removeToken(_from, _tokenId);
+    addToken(_to, _tokenId);
+    
+    require(!_safe || checkAndCallSafeTransfer(_from, _to, _tokenId, _data));
+
+    Transfer(_from, _to, _tokenId);
+  }
+
+  /**
+  * @dev Internal function to clear current approval of a given token ID
+  * @dev Reverts if the given address is not indeed the owner of the token
+  * @param _owner owner of the token
+  * @param _tokenId uint256 ID of the token to be transferred
+  */
+  function clearApproval(address _owner, uint256 _tokenId) internal {
+    require(ownerOf(_tokenId) == _owner);
+    if (tokenApprovals[_tokenId] != 0) {
+      tokenApprovals[_tokenId] = 0;
+      Approval(_owner, 0, _tokenId);
+    }
+  }
+
+  /**
+  * @dev Internal function to add a token ID to the list of a given address
+  * @param _to address representing the new owner of the given token ID
+  * @param _tokenId uint256 ID of the token to be added to the tokens list of the given address
+  */
+  function addToken(address _to, uint256 _tokenId) internal {
+    require(tokenOwner[_tokenId] == address(0));
+    tokenOwner[_tokenId] = _to;
+    ownedTokensCount[_to] = ownedTokensCount[_to].add(1);
+  }
+
+  /**
+  * @dev Internal function to remove a token ID from the list of a given address
+  * @param _from address representing the previous owner of the given token ID
+  * @param _tokenId uint256 ID of the token to be removed from the tokens list of the given address
+  */
+  function removeToken(address _from, uint256 _tokenId) internal {
+    require(ownerOf(_tokenId) == _from);
+    ownedTokensCount[_from] = ownedTokensCount[_from].sub(1);
+    tokenOwner[_tokenId] = 0;
+  }
+
+  /**
+  * @dev Internal function to invoke `onERC721Received` on a target address
+  * @dev The call is not executed if the target address is not a contract
+  * @dev Returns whether the call correctly returned the expected magic value
```

Comment:
Any reason why this wasn't implemented as throwing instead of returning a boolean? If the return value is kept it should be documented with a `@return` annotation.

Reply:
Good point, I'll change it to `throw` directly